assets/js/modules/adsense/components/common/AdBlockingRecoveryToggle.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * AdBlockingRecoveryToggle component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useMount } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { Switch } from 'googlesitekit-components';
+import {
+	AD_BLOCKING_FORM_SETTINGS,
+	MODULES_ADSENSE,
+} from '../../datastore/constants';
+import Link from '../../../../components/Link';
+import SettingsNotice from '../../../../components/SettingsNotice/SettingsNotice';
+import { useFeature } from '../../../../hooks/useFeature';
+import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function AdBlockingRecoveryToggle() {
+	const adBlockerDetectionEnabled = useFeature( 'adBlockerDetection' );
+
+	const adBlockerDetectionSnippet = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getUseAdBlockerDetectionSnippet()
+	);
+	const adBlockerDetectionErrorSnippet = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getUseAdBlockerDetectionErrorSnippet()
+	);
```

Comment:
For now, I have renamed the variables to `adBlocking*`.

Reply:
Hi, I've filled #7164 to address this. Also `adBlockingDetection` is another variation of inconsistent mismatch that we don't want. Maybe it's better to revert to `adBlockerDetection` since that's the selector names at the moment.